cookiecutter/repository.py

```diff
@@ -112,9 +113,14 @@ def determine_repo_dir(template, abbreviations, clone_to_dir, checkout,
         cleanup = False
 
     for repo_candidate in repository_candidates:
+        if dir:
+            repo_candidate = os.path.join(repo_candidate, dir)
         if repository_has_cookiecutter_json(repo_candidate):
             return repo_candidate, cleanup
 
+    if dir:
```

Comment:
Could we move this to before L115? This means we could drop the condition on L116...